exporter/splunkhecexporter/config.go

```diff
@@ -117,3 +152,56 @@ func (cfg *Config) getURL() (out *url.URL, err error) {
 
 	return
 }
+
+func (cfg *Config) GetSourceKey() string {
+	return cfg.HecMetadata.Source
+}
+
+func (cfg *Config) GetSourceTypeKey() string {
+	return cfg.HecMetadata.SourceType
+}
+
+func (cfg *Config) GetIndexKey() string {
+	return cfg.HecMetadata.Index
+}
+
+func (cfg *Config) GetHostKey() string {
+	return cfg.HecMetadata.Host
+}
+
+func (cfg *Config) GetNameKey() string {
+	return cfg.HecFields.Name
+}
+
+func (cfg *Config) GetSeverityTextKey() string {
+	return cfg.HecFields.SeverityText
+}
+
+func (cfg *Config) GetSeverityNumberKey() string {
+	return cfg.HecFields.SeverityNumber
+}
+
+// initialize the configuration
+func (cfg *Config) initialize() {
+	if cfg.HecMetadata.Source == "" {
+		cfg.HecMetadata.Source = splunk.DefaultSourceLabel
+	}
+	if cfg.HecMetadata.SourceType == "" {
+		cfg.HecMetadata.SourceType = splunk.DefaultSourceTypeLabel
+	}
+	if cfg.HecMetadata.Index == "" {
+		cfg.HecMetadata.Index = splunk.DefaultIndexLabel
+	}
+	if cfg.HecMetadata.Host == "" {
+		cfg.HecMetadata.Host = conventions.AttributeHostName
+	}
+	if cfg.HecFields.SeverityText == "" {
+		cfg.HecFields.SeverityText = splunk.DefaultSeverityTextLabel
+	}
+	if cfg.HecFields.SeverityNumber == "" {
+		cfg.HecFields.SeverityNumber = splunk.DefaultSeverityNumberLabel
+	}
+	if cfg.HecFields.Name == "" {
+		cfg.HecFields.Name = splunk.DefaultNameLabel
+	}
```

Comment:
Can't we put these to createDefaultConfig instead? I believe nested fields in default config can be overridden as well

Reply:
done